PR: https://github.com/odoo/odoo/pull/83274

From: a1904aa6f62acbd3d8e7d4cc873692e20edb37fd
From: Raphael Collet
Date: 2022-01-28 14:10:01

Structural Changes: 5
Total Changes: 6

[IMP] core: field index names

The possible index names have been renamed "btree", "btree_not_null"
(instead of "not null") and "trigram" (instead of "gin").

Task 2742526

Part-of: odoo/odoo#83274

================================= pseudo patch: =================================

--- a/addons/mail/models/mail_blacklist.py
+++ b/addons/mail/models/mail_blacklist.py
@@ -12,7 +12,7 @@ class MailBlackList(models.Model):
     _description = 'Mail Blacklist'
     _rec_name = 'email'
 
-    email = fields.Char(string='Email Address', required=True, index='gin', help='This field is case insensitive.',
+    email = fields.Char(string='Email Address', required=True, index='trigram', help='This field is case insensitive.',
                         tracking=True)
     active = fields.Boolean(default=True, tracking=True)
 

--- a/addons/mail/models/mail_message.py
+++ b/addons/mail/models/mail_message.py
@@ -96,7 +96,7 @@ class Message(models.Model):
         help='Attachments are linked to a document through model / res_id and to the message '
              'through this field.')
     parent_id = fields.Many2one(
-        'mail.message', 'Parent Message', index="not null", ondelete='set null',
+        'mail.message', 'Parent Message', index='btree_not_null', ondelete='set null',
         help="Initial thread message.")
     child_ids = fields.One2many('mail.message', 'parent_id', 'Child Messages')
     # related document

--- a/addons/mail/models/mail_shortcode.py
+++ b/addons/mail/models/mail_shortcode.py
@@ -13,7 +13,7 @@ class MailShortcode(models.Model):
 
     _name = 'mail.shortcode'
     _description = 'Canned Response / Shortcode'
-    source = fields.Char('Shortcut', required=True, index='gin', help="The shortcut which must be replaced in the Chat Messages")
+    source = fields.Char('Shortcut', required=True, index='trigram', help="The shortcut which must be replaced in the Chat Messages")
     substitution = fields.Text('Substitution', required=True, help="The escaped html code replacing the shortcut")
     description = fields.Char('Description')
     message_ids = fields.Many2one('mail.message', string="Messages", store=False)
